browser/ui/views/frame/brave_browser_frame.cc

```diff
@@ -74,3 +77,18 @@ void BraveBrowserFrame::SetTabDragKind(TabDragKind kind) {
         ->TabDraggingEnded(browser_view_->browser());
   }
 }
+
+ui::ColorProviderKey BraveBrowserFrame::GetColorProviderKey() const {
+  auto key = BrowserFrame::GetColorProviderKey();
+
+  if (base::FeatureList::IsEnabled(color::features::kBraveDarkerTheme)) {
+    if (!key.scheme_variant &&
+        key.color_mode == ui::ColorProviderKey::ColorMode::kDark &&
+        browser_view_->browser()->profile()->GetPrefs()->GetBoolean(
+            color::prefs::kBraveDarkerMode)) {
+      key.scheme_variant = ui::ColorProviderKey::SchemeVariant::kDarker;
```

Comment:
curious what it means if key_scheme_variant is non-null and our darker mode(`kBraveDarkerMode`) is enabled.

it means we don't overwrite if if upstream usea another variant?

scheme variant is very new to me so need to understand :)

Reply:
> it means we don't overwrite if if upstream usea another variant?

Yeah, at the moment that's what I intended. IIUC, scheme variant is used make variation on color scheme like, "dark, light, system", by populating palette dynamically. So I thought "Darker" theme is also be part of scheme variant.

And it's used in Ash, practically we wouldn't bump into it.

Reply:
Actually I was wrong. When using custom theme, this value will be filled per color. I think we should overwrite the value.